src/main/java/org/opensearch/security/dlic/rest/api/InternalUsersApiAction.java

```diff
@@ -228,6 +241,37 @@ ValidationResult<SecurityConfiguration> validateSecurityRoles(final SecurityConf
         );
     }
 
+    /**
+     * This method combines roles from both 'opensearch_security_roles' and the deprecated 'opendistro_security_roles' fields.
+     * If the deprecated field is used, a warning message is logged.
+     *
+     * @param content The SecurityJsonNode containing the security configuration
+     * @return A List of merged unique roles. Returns an empty list if no roles are found in either field.
+     */
+    List<String> getRoles(SecurityJsonNode content) {
+        List<String> openSearchRoles = content.get(DIRECT_SECURITY_ROLES).asList();
+        List<String> openDistroRoles = content.get(OPENDISTRO_SECURITY_ROLES).asList();
+
+        Set<String> mergedRoles = new HashSet<>();
+
+        // Add OpenSearch roles if present
+        if (openSearchRoles != null) {
+            mergedRoles.addAll(openSearchRoles);
+        }
+
+        // Add OpenDistro roles if present
+        if (openDistroRoles != null) {
+            mergedRoles.addAll(openDistroRoles);
+            log.warn(
```

Comment:
Indirectly related: What's the migration path towards having only the new attribute?

This will only log warnings when the REST API is used. No warnings will be logged for usage of the securityadmin tool or just the existing data in the config index.

Will there be an additional change in the future to address existing data?

Reply:
Can we do a clean break with a deprecation message that clearly states that this will be remove in 3.0 GA?